local-dmabuf.c

```diff
@@ -78,13 +116,24 @@ local_create_dmabuf(struct iio_buffer_pdata *pdata, size_t size, void **data)
 		.fd_flags = O_CLOEXEC | O_RDWR,
 	};
 	struct iio_block_pdata *priv;
+	const char *heap_name;
+	char dma_heap_path[MAX_DMA_HEAP_PATH];
 	int ret, fd, devfd;
 
 	priv = zalloc(sizeof(*priv));
 	if (!priv)
 		return iio_ptr(-ENOMEM);
 
-	devfd = open("/dev/dma_heap/system", O_RDONLY | O_CLOEXEC | O_NOFOLLOW); /* Flawfinder: ignore */
+	/* Determine DMA heap path from global environment variable */
+	heap_name = get_dma_heap_name();
+
+	ret = snprintf(dma_heap_path, sizeof(dma_heap_path), "/dev/dma_heap/%s", heap_name);
```

Comment:
If we go the path of defaulting to "system" in case of error, you can just call `get_dma_heap_name()` inline in `snprintf()`